DigitalLearningSolutions.Web/ServiceFilter/VerifyAdminUserCanViewCourse.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace DigitalLearningSolutions.Web.ServiceFilter
+{
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Helpers;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Mvc.Filters;
+
+    public class VerifyAdminUserCanViewCourse : IActionFilter
+    {
+        private readonly ICourseService courseService;
+
+        public VerifyAdminUserCanViewCourse(
+            ICourseService courseService
+        )
+        {
+            this.courseService = courseService;
+        }
+
+        public void OnActionExecuted(ActionExecutedContext context) { }
+
+        public void OnActionExecuting(ActionExecutingContext context)
+        {
+            if (!(context.Controller is Controller controller))
+            {
+                return;
+            }
+
+            var centreId = controller.User.GetCentreId();
+            var categoryId = controller.User.GetAdminCategoryId()!;

```

Comment:
Could this filter ever be used in a context where this `categoryId` could actually be null?

I guess this would only happen if a non-admin user attempted to view something in which case anything that kicks them out is helpful - but a controlled exit would be better.

Reply:
I think this is a case where the order of attributes is important. I believe the Authorize attribute will throw out any non-admin users before we get to these VerifyAdminCan service filters (assuming we set up the controllers properly).